### PR TITLE
[ME-11] implements upcoming jobs proximity processor

### PR DESCRIPTION
**Pull request by venkatesh-thalluri**

### CODE REVIEW

File: src/main/java/com/dmg/matchingengine/model/entity/PropertyEntity.java [hunk 1]
```diff
@@ -1,15 +1,31 @@
...
 @Entity
 @Table(name = "property")
+@Slf4j
 public class PropertyEntity extends BaseEntity {
+    @Column(name = "location")
+    private String location;
 
+    public Location getLocation() {
+        try {
+            return new ObjectMapper().readValue(this.location, Location.class);
```
Comment (dmittal-dmg) [inline src/main/java/com/dmg/matchingengine/model/entity/PropertyEntity.java:25]: Import Mapper in commons or create a singleton object maper. Do not create new instances.

File: src/main/java/com/dmg/matchingengine/model/entity/PropertyEntity.java [hunk 2]
```diff
@@ -1,15 +1,31 @@
...
 import javax.persistence.Entity;
 import javax.persistence.Table;
 
 @Getter
 @Setter
 @Entity
 @Table(name = "property")
+@Slf4j
 public class PropertyEntity extends BaseEntity {
+    @Column(name = "location")
```
Comment (dmittal-dmg) [inline src/main/java/com/dmg/matchingengine/model/entity/PropertyEntity.java:20]: Can we store lat and long seperatly?

File: src/main/java/com/dmg/matchingengine/model/entity/PropertyEntity.java [hunk 3]
```diff
@@ -1,15 +1,31 @@
...
+    private String location;
 
+    public Location getLocation() {
+        try {
+            return new ObjectMapper().readValue(this.location, Location.class);
+        } catch (JsonProcessingException e) {
+            log.error("error occurred while converting json string to location", e);
+        }
+        return new Location("default", 0, 0);
```
Comment (dmittal-dmg) [inline src/main/java/com/dmg/matchingengine/model/entity/PropertyEntity.java:29]: Checking with Manik, what if the coordinates arent available. What should be the default bucket.

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -49,6 +49,13 @@ spring:
         jdbc:
           time_zone: ${JPA_PROPERTIES_JDBC_TIME_ZONE}
           batch_size: ${JPA_PROPERTIES_JDBC_BATCH_SIZE}
+      javax:
+        persistence:
+          schema-generation:
```
Comment (dmittal-dmg) [inline src/main/resources/application.yml:54]: What does this do?

Comment (venkatesh-thalluri) [reply src/main/resources/application.yml:54]: this is for my internal to create schema. removed this. not needed here

File: src/main/java/com/dmg/matchingengine/model/entity/PropertyEntity.java (hunk 2 above)
Comment (venkatesh-thalluri) [reply src/main/java/com/dmg/matchingengine/model/entity/PropertyEntity.java:20]: lat and long on separate in tis string (json) as this part is not yet decided kept it this way. Can update it to lat and long columns

File: src/main/java/com/dmg/matchingengine/model/entity/PropertyEntity.java (hunk 3 above)
Comment (venkatesh-thalluri) [reply src/main/java/com/dmg/matchingengine/model/entity/PropertyEntity.java:29]: there can be no default as per Manik. He is asking to throw an error and that error needs to be appeared in datadog. adapted changes accordingly

File: src/main/java/com/dmg/matchingengine/model/entity/PropertyEntity.java (hunk 1 above)
Comment (venkatesh-thalluri) [reply src/main/java/com/dmg/matchingengine/model/entity/PropertyEntity.java:25]: no longer needed. removed it

**Review by dmittal-dmg — APPROVED** [review]

**Pull request merged**